app/assets/javascripts/admin/addon/components/color-palette-editor.gjs

```diff
@@ -78,28 +54,58 @@ const Picker = class extends Component {
     const color = event.target.value.replace("#", "");
     if (this.args.showDark) {
       this.args.onDarkChange(color);
-      this.args.color.darkValue = color;
     } else {
       this.args.onLightChange(color);
-      this.args.color.lightValue = color;
     }
   }
 
   @action
   onTextChange(event) {
-    const color = event.target.value;
+    let color = event.target.value;
+
+    if (!this.isValidHex(color)) {
+      event.preventDefault();
+      return;
+    }
+
+    color = this.ensureSixDigitsHex(color);
     if (this.args.showDark) {
       this.args.onDarkChange(color);
-      this.args.color.darkValue = color;
     } else {
       this.args.onLightChange(color);
-      this.args.color.lightValue = color;
     }
   }
 
   @action
   onTextKeypress(event) {
-    const color = event.target.value + event.key;
+    const currentValue = event.target.value;
+
+    if (event.keyCode === 13) {
+      event.preventDefault();
+
+      if (currentValue.length !== 6 && currentValue.length !== 3) {
```

Comment:
Instead of having to manually validate this, I think we can just rely on the `minLength` and `maxLength` attributes on the `input` to restrict the length?

Reply:
Also, it has limitation that toast will show only when admin hit enter (13 keyCode). But I can click outside and will never know that something is wrong.

Demo

https://github.com/user-attachments/assets/5e13dffc-7e35-4fec-ab18-aadfb90656bb

When I am out of focus and value is invalid, should we display error as well?

Reply:
I don't believe `minlength` and `maxlength` would work here. We want to allow values that are either exactly 3 OR 6 chars long; 4 and 5 are illegal.

Reply:
@lis2 I've pushed a change to show an error toast if the value is invalid when moving focus outside the text input (and show red border around inputs containing invalid colors).